doc/adr/0010-data-types-in-columns.md

```diff
@@ -0,0 +1,137 @@
+    /**
+     * Data is textual data
+     */
+    STRING,
+    /**
+     * Data representing a time range of string: [start,end],
+     * where `start` and `end` are timestamps in nanoseconds
+     */
+    TIME_RANGE;
+}
+eclipse-cdt-cloud
+```
+
+### Use cases
+
+The first concrete use case for the data types is the `Select min/max time range` in Trace Compass' latency views. New columns can be provided with the min and max time range. The Column descriptor will provide the `data type` and the cell value will transport the range as string `[start,end]`. To select the time range the following user actions can be implemented in the front-end:
+
+- Make the cell entry "clickable", similar to hyperlinks
+  - Advantages:
+    - More discoverable for the user if column is visible on screen and not off screen
+    - Addtional actions can be implemented (CTRL + click or conext-sensitive menu)
+  - Disadvantages:
+    - These columns need to be visible for the feature
+- Create a context sensitive-menu per row with menu items per `TIME_RANGE` column. The user can select one to apply the time range as selection. This is the way how Eclipse Trace Compass provides the feature.
+  - Advantages:
+    - Columns can be made hidden.
+  - Disadvantages:
+    - Context-sensitive menu is less discoverable
+
+Notes:
+
+- Both options can be implemented at the same time
+- Action can be made configurable (e.g. `Select and navigate` instead of `Select`). Both options can co-exist.
+
+### Example sequence siagram
+
+Prerequiste `Open Trace and fetched property columndescriptors` for all diagrams:
+
+```mermaid
+sequenceDiagram
+    actor User
+    participant client as Theia Trace
+    participant server as Trace Server
+    User->>client: open an experiment
+    client->>server: POST /experiments/
+    server->>client: 200: Experiment
+    client->>server: GET /experiment/{expUuid}/outputs
+    server->>client: 200: OutputDescriptor[]
+   
+```
+
+#### Go To Max
+
+The following feature exists in Eclipse Trace Compass for all latency statistics tables. It provides the possibility to navigate to the time range where the maximum latency occured. The metadata give some additional information that can be used in the front-end to identify UI elements (not available in Eclipse Trace Compass).
+
+```mermaid
+sequenceDiagram
+    actor User
+    participant client as Theia Trace
+    participant server as Trace Server
+    Note over client,server: Prerequisite Open Trace and fetched property descriptors
+    User->>client: open Latency Table
+    client->>server: POST /experiment/{expUuid}/outputs/{outputId}/data/tree
+    server->>client: 200: TreeModel
+    Note over client,server: TreeModel incl. data type and time range columns
+    client->>client: populate view
+    User->>client: right-mouse-click on row
+    loop foreach time range column
+        client->>client: create menu entry based on time range column
+    end
+    User->>client: select "Go To Max" menu item
```

Comment:
Following your proposal, is the following statement correct?

- only tables with a column of type "TIME_RANGE" will enable the client (theia-trace-ext) to implement the go to min max feature

Reply:
yes. Corresponding time range columns need to exist.

Reply:
How does the client know that a specific "TIME_RANGE" column represent a column with the "**max** time ranges"? I.e. can the UI understand that a specific column contain time ranges of a specific kind (**max** in this case)?

Reply:
The time range column "tells" the client that a time range can be selected (or navigated to). The actual name is determined by the column name. So, for a column `Max Time Range` the action will `Select Max Time Range`, i.e `Select` is prepended to `Max Time Range`.

Reply:
Based on this conversation, should the label be: select "Select (column name)" menu item? Or: select "Go To (column name)" menu item?

Reply:
Done